File: solutions/maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.cpp
// @Title: Maximum Depth of Binary Tree
// @Author: colinjxc
// @Date: 2022-02-06T14:41:11+08:00
// @URL: https://leetcode-cn.com/problems/maximum-depth-of-binary-tree


/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        if (root == nullptr) return 0;
        return std::max(maxDepth(root->right), maxDepth(root->left)) + 1;
    }
};

File: solutions/maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.go
// @Title: Maximum Depth of Binary Tree
// @Author: colinjxc
// @Date: 2021-11-02T12:30:36+08:00
// @URL: https://leetcode-cn.com/problems/maximum-depth-of-binary-tree


/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func maxDepth(root *TreeNode) int {
    if root ==  nil {
        return 0
    }
    return max(maxDepth(root.Left), maxDepth(root.Right)) + 1
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b;
}
